File: src/db-gwent-server-scala/src/main/scala/com/entities/abilities/RemoveWeather_Ability.scala
package com.entities.abilities

import com.entities.{CardRowType, Player, Card}

case class RemoveWeather_Ability()
  extends Ability:

  def apply(card: Card, player: Player, enemy: Player): Boolean =
    Array(player, enemy).forall:
      p => p.removeCardModifiersOfId(RowCardsToOne_Ability.Id)

File: src/db-gwent-server-kotlin/src/main/kotlin/com/entities/abilities/RemoveWeather_Ability.kt
package com.entities.abilities

import com.entities.Card
import com.entities.Player

class RemoveWeather_Ability : Ability {

    override fun apply(card: Card, thisPlayer: Player, nextPlayer: Player): Boolean =
        arrayOf(thisPlayer, nextPlayer).all{
            it.removeCardModifiersOfId(RowCardsToOne_Ability.Id)
        }

}
